packages/lib/src/vis/tiles/TiledHeatmapLayer.tsx

```diff
@@ -0,0 +1,72 @@
+import { Suspense } from 'react';
+import { LinearFilter, NearestFilter, Vector2 } from 'three';
+
+import { useVisibleDomains } from '../hooks';
+import { useAxisSystemContext } from '../shared/AxisSystemContext';
+import Tile from './Tile';
+import type { TilesApi } from './api';
+import type { ColorMapAttributes } from './models';
+import { getTileOffsets, sortTilesByDistanceTo } from './utils';
+
+interface Props extends ColorMapAttributes {
+  api: TilesApi;
+  lod: number;
+}
+
+function TiledHeatmapLayer(props: Props) {
+  const { api, lod, ...colorMapAttrs } = props;
+
+  const nLevels = api.levelsSize.length;
+  const [level0Size] = api.levelsSize;
+  const { width, height } = api.levelsSize[lod];
+  const { tileSize } = api;
+
+  const { xVisibleDomain, yVisibleDomain } = useVisibleDomains();
+  const { visSize } = useAxisSystemContext();
+
+  // Transform visible domain to current level-of-detail array coordinates
+  const xScale = width / level0Size.width;
+  const yScale = height / level0Size.height;
+  const origin = new Vector2(
+    Math.max(0, xVisibleDomain[0] * xScale),
+    Math.max(0, yVisibleDomain[0] * yScale)
+  );
+  const end = new Vector2(
+    Math.min(width, xVisibleDomain[1] * xScale),
+    Math.min(height, yVisibleDomain[1] * yScale)
+  );
+  const tileOffsets = getTileOffsets(origin, end, tileSize);
+
+  // Sort tiles from closest to vis center to farthest away
+  const center = new Vector2(
+    0.5 * (origin.x + end.x),
+    0.5 * (origin.y + end.y)
+  );
+  sortTilesByDistanceTo(tileOffsets, tileSize, center);
+
+  return (
+    <group
+      position={[
+        -visSize.width / 2,
+        -visSize.height / 2,
+        (nLevels - lod) / (nLevels + 1),
+      ]}
+      scale={[visSize.width / width, visSize.height / height, 1]}
```

Comment:
I think it is worth keeping the comment saying that this is needed to work in lod coordinates